components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java

```diff
@@ -143,14 +143,20 @@ public Response deleteRegistry(String registryId, MessageContext messageContext)
 
     @Override
     public Response getRegistryEntryByUuid(String registryId, String entryId, MessageContext messageContext) {
-        RegistryEntryDTO registryEntry = new RegistryEntryDTO();
-        registryEntry.setEntryName("Pizzashack-endpoint");
-        registryEntry.setMetadata("{ \"mutualTLS\" : true }");
-        registryEntry.setDefinitionType(RegistryEntryDTO.DefinitionTypeEnum.OAS);
-        registryEntry.setDefinitionUrl("http://localhost/pizzashack?swagger.json");
-        registryEntry.setServiceType(RegistryEntryDTO.ServiceTypeEnum.REST);
-        registryEntry.setServiceUrl("http://localhost/pizzashack");
-        return Response.ok().entity(registryEntry).build();
+        EndpointRegistry registryProvider = new EndpointRegistryImpl();
+        EndpointRegistryEntry endpointRegistryEntry = null;
+        try {
+            endpointRegistryEntry = registryProvider.getEndpointRegistryEntryByUUID(entryId);
```

Comment:
Shouldn't we verify the existence of the registry before getting the entry? Otherwise, this might work when querying using wrong registry id and correct entry id.

Or else, we should change the resource without requiring the registry id right?

Reply:
correct. will fix it.

Reply:
fixed